lib/environment.js

```diff
@@ -0,0 +1,57 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+/**
+ * @private
+ * @return {boolean} True, if we're running in the service worker global scope.
+ * False otherwise.
+ */
+function isServiceWorkerGlobalScope() {
+  return ('ServiceWorkerGlobalScope' in self &&
+          self instanceof ServiceWorkerGlobalScope);
+}
+
+/**
+ * @private
+ * @return {boolean} True, if we're running a development bundle.
+ * False otherwise.
+ */
+function isDevBuild() {
+  // The string `BUILD_TARGET` will be replaced during the build process.
+  return `BUILD_TARGET` === `dev`;
```

Comment:
Nit: Could you add some kind of prefix along the lines of BUILD_PROCESS_REPLACE::BUILD_TARGET

Just so it's more obvious what is happening here.